### PR TITLE
ARM DNS commands implemented

### PR DESCRIPTION
**Pull request by MikhailTryakhov**
The content to be added to Changelog is as follows:
* Moved to new api version 2016-04-01
* Implemented new ```dns zone clear``` command to remove all related record sets
* ```dns zone delete``` command now removes dns zone and all related record sets
* Implemented command ```dns record-set set-soa-record''' to set SOA type record
* Records of type PTR is now supported
* Added functionality to create dns zone, record sets from ```record-set add-record``` command.
* Reworked ```dns zone list``` command to make ```resource-group``` optional parameter
* DNS record sets now supports ```metadata``` parameter instead of ```tags```
* Reworked TXT records functionality. Max record length is 1024, splitted by 255 simbols.
* DNS zone properties now include **nameServers**
* Record sets have no **location** property
* SOA record have an additional **serialNumber** field
* Fixed issue [#1795](https://github.com/Azure/azure-xplat-cli/issues/1795)

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
Hi @amarzavery 
could you please review my PR with huge improvements to network DNS services.

File: lib/commands/arm/network/dnsZone._js [hunk 1]
```diff
@@ -307,60 +352,90 @@ __.extend(DnsZone.prototype, {
...
   },
 
   deleteRecordSet: function (resourceGroupName, zoneName, setName, options, _) {
     var self = this;
-    options.type = self._validateType(options.type);
 
-    if (!options.quiet && !self.interaction.confirm(util.format($('Delete DNS record set "%s"? [y/n] '), setName), _)) {
+    var recordSet = self.getRecordSet(resourceGroupName, zoneName, setName, options.type, _);
+    if(!recordSet) {
+      throw new Error(util.format($('A record set with name "%s" of type "%s" not found in the DNS zone "%s"'), setName, options.type, zoneName));
```
Comment (amarzavery) [inline lib/commands/arm/network/dnsZone._js]: Are you sure that options.type will always exist over here. We dont want the error message to say "of type undefined"

File: lib/commands/arm/network/network._js [hunk 2]
```diff
@@ -1588,19 +1603,41 @@ exports.init = function (cli) {
       dnsZone.setRecordSet(resourceGroup, dnsZoneName, name, options, _);
     });
 
-  dnsRecordSet.command('list [resource-group] [dns-zone-name] [type]')
+  dnsRecordSet.command('set-soa-record [resource-group] [dns-zone-name] [record-set-name]')
+    .description($('Set a SOA record in a record set under a DNS zone'))
+    .usage('[options] <resource-group> <dns-zone-name> <record-set-name>')
```
Comment (haocs) [inline lib/commands/arm/network/network._js]: Is `record-set-name` a required argument?

File: lib/commands/arm/network/dnsZone._js (hunk 1 above)
Comment (MikhailTryakhov) [inline lib/commands/arm/network/dnsZone._js]: Yes, options. type is required parameter here. We display prompt to enter such value.

File: lib/commands/arm/network/network._js (hunk 2 above)
Comment (MikhailTryakhov) [inline lib/commands/arm/network/network._js]: Correct, it's required argument in this command

**Pull request merged**